autonity/autonity.go

```
@@ -92,8 +92,7 @@ func (ac *Contract) MeasureMetricsOfNetworkEconomic(header *types.Header, stateD
 	}
 
 	// marshal the data from bytes arrays into specified structure.
-	v := EconomicMetaData{make([]common.Address, 32), make([]uint8, 32), make([]*big.Int, 32),
-		make([]*big.Int, 32), new(big.Int), new(big.Int)}
+	v := EconomicMetaData{make([]common.Address, 32), make([]uint8, 32), make([]*big.Int, 32), new(big.Int), new(big.Int)}
```

Hi @Jason-Zhangxin-Chen thanks for the test, I'm looking at this more closely and thinking that actually this function should just return an error if there is a problem. Currently it just swallows any error encountered and I'm not sure why. But if it returned an error we could be sure that the metrics are always working, maybe we wouldn't even need the test you just added.

---

> maybe we

Yes, may be you are right.
May be the oringinal thinking was don't let the metric measurement to block the commit of the block, so it just log the warning.

---

In this situation, given that we are patching a release to fix this, it seems that the metrics have sufficient importance to block the commit of a block.

So I think we should go ahead with causing the block creation to fail if we encounter an error in the processing of the metrics.